### PR TITLE
Adds webresource typing generation

### PR DESCRIPTION
**Pull request by otaviojacobi**
Change-type: minor

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -29,6 +29,13 @@ export type Deferred<T> = Exclude<T, any[]>;
 export type PickDeferred<T, K extends keyof T> = {
 	[P in K]: Deferred<T[P]>;
 };
+export interface WebResource {
```
Comment (otaviojacobi) [inline src/index.ts:32]: @Page- Previous version of this was using a direct "{}" for the WebResource type. Although I see that it is a bit odd to have this aditional type (non-native like Date or Buffer) if I was the main user of this library I would like to have the type checking on a WebResource type, please let me know what do you think.

As for the 'write' typing above I am afraid I don't really understand what it does - My search seems that pine only uses the read types ([here](https://github.com/balena-io/pinejs/blob/9acd11e1ccfc7047c07e29258290d9a1f49b3018/src/bin/abstract-sql-compiler.ts#L25) and [here](https://github.com/balena-io-modules/abstract-sql-to-typescript/blob/c758673bd7a88923f1ce5f4f4aed2cf423d88f29/src/index.ts#L229))

File: package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
   "repository": "https://github.com/balena-io-modules/abstract-sql-to-typescript.git",
   "author": "",
   "dependencies": {
-    "@balena/abstract-sql-compiler": "^8.0.0",
-    "@balena/odata-to-abstract-sql": "^5.9.1",
+    "@balena/abstract-sql-compiler": "8.0.1-build-bumps-sbvr-types-with-web-resource-73ce959674cda5f0a89dabd4954d0eafcb1e366b-1",
```
Comment (otaviojacobi) [inline package.json:16]: TODO: clean this up when proper down-stream is merged

File: src/index.ts (hunk 1 above)
Comment (Page-) [reply src/index.ts:32]: The write typing is what you can use for `.post`/`.patch` bodies vs the read typing which is what you get back from a `.get`. Usually they're the same but there can be some differences, the most notable being that reading a foreign key (object) is very different to writing it (number/etc). The difference could do with being more fleshed out but this module is only barely past being a PoC so the same could be said about most of it, it's still very handy though

**Comment by otaviojacobi** [comment]
LGTM

**Pull request merged**